src/build-info/src/lib.rs

```diff
@@ -98,6 +98,7 @@ macro_rules! build_info {
                            printf "either build from working Git clone " >&2
                            printf "(see https://materialize.com/docs/install/#build-from-source), " >&2
                            printf "or specify SHA manually by setting MZ_DEV_BUILD_SHA environment variable" >&2
+                           printf "If you are using git worktrees, you must be in the primary worktree." >&2
```

Comment:
You can also do what's written in the previous line, `export MZ_DEV_BUILD_SHA` worked ok for me.

Reply:
Yeah, I mostly just want it clear that the worktrees are _why_ the automatic detection is not working. Do you have a suggested rephrasing to make that more clear?

Reply:
I just pushed a slight rephrasing.